stdlib/REPL/src/REPL.jl

```diff
@@ -73,6 +73,24 @@ mutable struct REPLBackend
         new(repl_channel, response_channel, in_eval)
 end
 
+function softscope!(ex)
+    if ex isa Expr
+        h = ex.head
+        if h === :toplevel
+            for i = 1:length(ex.args)
+                ex.args[i] = softscope!(ex.args[i])
+            end
+        elseif h in (:meta, :import, :using, :export, :module, :error, :incomplete, :thunk)
+            return ex
+        else
+            return Expr(:block, Expr(:softscope, true), ex)
+        end
+    end
+    return ex
+end
+
+const repl_ast_transforms = Any[softscope!]
```

Comment:
Hopefully, this will allow OhMyREPL.jl and Revise.jl and other packages that require REPL hooks to play nicely, since they can just push new transforms to `REPL.repl_ast_transforms`.   But I'm not sure if it should be a global variable or should be a property specific to a REPL instance?

See also the discussion in stevengj/SoftGlobalScope.jl#2; cc @dawbarton, @KristofferC, @timholy.

Reply:
Does seem like it should be per-REPL-instance.

Reply:
I agree it should be per-REPL, but I'm not sure how to implement it. E.g. `LineEditREPL` does not directly contain a `REPLBackend`, so passing around which transforms to use is nontrivial. Hopefully somebody can help with that later.